### PR TITLE
[IMP] Add hook on the create settlements

### PR DESCRIPTION
**Pull request by etobella**
This PR adds a hook in order to create other systems.

### CODE REVIEW

File: sale_commission/models/account_invoice.py [hunk 1]
```diff
@@ -221,3 +221,15 @@ def _compute_company(self):
...
     ]
+
+    @api.multi
+    def _skip_settlement(self):
+        """
+        This function should return if the commission can be payed
+        :return: bool
+        """
+        self.ensure_one()
+        return bool(
```
Comment (pedrobaeza) [inline sale_commission/models/account_invoice.py]: No need to cast to bool. It's already a boolean.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
We have agent types for discriminating the payment of settlements, but I don't get why do you want to skip settlements at all.

**Comment by etobella** [comment]
It could be used for the following: 

- Pay agents when the invoice is payed or when the due_date has been accomplished.
- Use the same formula for all agents (in order to avoid duplicity) but you want that some agents are payed as soon as it is invoiced.
- You want to create a different way to manage when to pay the invoices

**Comment by pedrobaeza** [comment]
> Pay agents when the invoice is payed or when the due_date has been accomplished.

OK, good use case, but I think this one should be included in the base module (that's to say, adding conditions in the method you have created). Don't you think?

> Use the same formula for all agents (in order to avoid duplicity) but you want that some agents are payed as soon as it is invoiced.

OK, the same about integrating this through an option at agent level.

> You want to create a different way to manage when to pay the invoices

I don't understand this one, as this is not for paying, but for settling the commissions.

**Comment by etobella** [comment]
The problem is that the system does not block settlements. When you create the settlement, it will be invoiced as soon as you launch the wizard.
So, if you want to block some `account.invoice.line.agent` to be payed, you should not allow to create the settlement. If your company has some conditions in order to decide when to create them, you use this hook in order to define it.

**Comment by pedrobaeza** [comment]
Well, you can do the following:

- Invoice the settlements you want selecting them and clicking on the action.
- Add a new settlement state/check for blocking them.

But now I understand what you want to achieve in that third point.

**Review by pedrobaeza — APPROVED** [review]

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**